plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/state/CodeModernizerState.kt

```diff
@@ -39,23 +38,24 @@ fun buildState(context: CodeModernizerSessionContext, isJobOngoing: Boolean, job
     )
 }
 
-fun CodeModernizerState.getLatestJobId() = JobId(lastJobContext[JobDetails.LAST_JOB_ID] ?: throw RuntimeException("No Job has been executed!"))
-
 class CodeModernizerState : BaseState() {
     @get:Property
     val lastJobContext by map<JobDetails, String>()
 
     @get:Property
     val flags by map<StateFlags, Boolean>()
-}
-
-enum class JobDetails {
-    LAST_JOB_ID,
-    CONFIGURATION_FILE_PATH,
-    TARGET_JAVA_VERSION,
-    SOURCE_JAVA_VERSION,
-}
 
-enum class StateFlags {
-    IS_ONGOING
+    fun getLatestJobId() = JobId(lastJobContext[JobDetails.LAST_JOB_ID] ?: throw RuntimeException("No Job has been executed!"))
+
+    fun toSessionContext(project: Project): CodeModernizerSessionContext {
+        val configurationFile = lastJobContext[JobDetails.CONFIGURATION_FILE_PATH]?.toVirtualFile()
+            ?: throw RuntimeException("No build file store in the state")
```

Comment:
normally we would just do `error` but i see that this was a move so no need to fix